qiling/os/posix/syscall.py

```diff
@@ -252,24 +251,29 @@ def ql_syscall_openat(ql, openat_fd, openat_path, openat_flags, openat_mode, nul
     openat_flags = openat_flags & 0xffffffff
     openat_mode = openat_mode & 0xffffffff
 
-    if os.path.exists(real_path) == False:
+    for i in range(256):
+        if ql.file_des[i] == 0:
+            idx = i
+            break
+
+    if idx == -1:
```

Comment:
Does it need to initialize `idx` with `-1` first?

Reply:
u r right, fixed it with an extra `else` to do that

Reply:
The `else` statement added is odd: missing indentation.
By the way, maybe you can refer to this:
https://github.com/qilingframework/qiling/blob/6596ceb1997adbff04f7add1f7012ba7d6adbb65/qiling/os/posix/syscall.py#L1242-L1254

Reply:
the `else` here is paired with `for` statement so the indentation is fine

Reply:
Sorry for that. Cool! Did't notice this before ...

Reply:
its okay, im glad you spot the oddity of idx value thing. thx